src/App.jsx

```diff
@@ -1,87 +1,98 @@
+    { value: "*", type: "operator" },
+    { value: "4", type: "number" },
+    { value: "5", type: "number" },
+    { value: "6", type: "number" },
+    { value: "-", type: "operator" },
+    { value: "1", type: "number" },
+    { value: "2", type: "number" },
+    { value: "3", type: "number" },
+    { value: "+", type: "operator" },
+    { value: "0", type: "number" },
+    { value: ".", type: "decimal" },
+    { value: "=", type: "action" },
   ];
-  const operators = ["+", "-", "*", "/"];
 
-  const handleButtonPress = (buttonValue) => {
-    if (!isNaN(buttonValue)) {
-      //Number button
-      setDisplayValue(
-        displayValue === "0" ? buttonValue : displayValue + buttonValue
-      );
-    } else if (operators.includes(buttonValue)) {
-      //Operator button
-      setPreviousValue(displayValue);
-      setOperator(buttonValue);
-      setDisplayValue(operator.toString());
-    } else if (buttonValue === "C") {
-      //Clear button
-      setDisplayValue("0");
-      setPreviousValue("");
-      setOperator("");
-    } else if (buttonValue === "=") {
-      //Equals button
-      const num1 = Number(previousValue);
-      const num2 = Number(displayValue);
-      let result;
-
-      if (operator === "+") {
-        result = num1 + num2;
-      } else if (operator === "-") {
-        result = num1 - num2;
-      } else if (operator === "*") {
-        result = num1 * num2;
-      } else if (operator === "/") {
-        result = num2 === 0 ? "Error" : num1 / num2;
+  const handleButtonPress = ({ value, type }) => {
+    if (display.length < maxDisplayLength) {
+      if (type === "operator" && display === "0") {
+        setDisplay(value);
+      } else if (type === "number") {
+        setDisplay(display === "0" ? value : display + value);
+      } else if (type === "operator") {
+        if (!isNaN(lastChar)) {
+          setDisplay(display + value);
+        } else if (lastChar && type === "operator") {
+          setDisplay(display.slice(0, -1) + value);
+        }
+      } else if (type === "decimal") {
+        if (lastChar === ".") {
+          return;
+        } else if (isNaN(lastChar)) {
+          setDisplay(display + "0" + value);
+        } else {
+          setDisplay(display + value);
+        }
       }
+    }
+  };
 
-      if (result !== undefined) {
-        setDisplayValue(result.toString());
-        setShowResult(true);
-        setPreviousValue("");
-        setOperator("");
-      } else {
+  const handleActions = ({ value, type }) => {
+    if (type === "action") {
+      if (value === "AC") {
+        setDisplay("0");
         setShowResult(false);
+      } else if (value === "C") {
+        if (display === "0") {
+          return;
+        } else if (display.length === 1) {
+          setDisplay("0");
+        } else {
+          setDisplay(display.slice(0, -1));
+        }
+      } else if (value === "=") {
+        {
+          setDisplay(display === "0" ? "0" : eval(display).toString());
```

Comment:
This eval is not really recommended, but replacing it is a bit of work. The way you implement the remaining logic, you'll probably need to use string splitting or regex to do it.